### ISSUE TITLE
how to sft?

### ISSUE THREAD

**Issue by wang-kangkang** [baaivision/Emu3#57]
i see the sft script:https://github.com/baaivision/Emu3/blob/f6ea9aaa44bd4ae5d7a048300cf0a16a863b0cae/scripts/t2i_sft.sh#L11
but the data format is not show. how to organize the train data format?

**Comment by ryanzhangfan** [comment]
please refer to https://github.com/baaivision/Emu3/blob/main/emu3/train/prepare_data.py and https://github.com/baaivision/Emu3/blob/main/emu3/train/datasets.py

**Comment by JuanJia** [comment]
> please refer to https://github.com/baaivision/Emu3/blob/main/emu3/train/prepare_data.py and https://github.com/baaivision/Emu3/blob/main/emu3/train/datasets.py

Hello, after completing the SFT (Supervised Fine-Tuning), I noticed that the model's size has reduced. Can you help me analysis the reason?
Only the following files were saved: 

- `model-00001-of-00004.safetensors`
- `model-00002-of-00004.safetensors`
- `model-00003-of-00004.safetensors`
- `model-00004-of-00004.safetensors`

![Image](https://github.com/user-attachments/assets/6a9e7c6f-09b2-42ff-9a5b-9af157d4c728)

**Comment by ryanzhangfan** [comment]
@JuanJia Please check the params number and dtype. The saved checkpoint might be in bf16, which occupies approximately 16G of disk space.

**Comment by JuanJia** [comment]
> [@JuanJia](https://github.com/JuanJia) Please check the params number and dtype. The saved checkpoint might be in bf16, which occupies approximately 16G of disk space.

To load the SFT checkpoint, we need to add certain .py files from the official weights folder, just copy configuration_emu3.py and modeling_emu3.py?

**Comment by ryanzhangfan** [comment]
Yes. It's the same as loading the official weights since they are exactly the "same" model.

**Comment by JuanJia** [comment]
The inference result of sft model is <|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|>. Can you help me analysis the reason?
